app/main.py

```diff
@@ -1 +1,19 @@
-# write your code here
+from app.errors import NotWearingMaskError, VaccineError
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> str:
+
+    masks_to_buy = 0
+    for person in friends:
+        try:
+            cafe.visit_cafe(person)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_to_buy += 1
+
+    if masks_to_buy:
+        return f"Friends should buy {masks_to_buy} masks"
+
+    return f"Friends can go to {cafe.name}"
```

Comment:
Code Efficiency: You can improve the efficiency of your code by not returning immediately when a VaccineError occurs. Instead, you could keep track of the number of friends who are not vaccinated and then return a more detailed message at the end. For example, 'x friends are not vaccinated'. This gives more information and does not stop the loop prematurely.